<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>947</width>
    <height>621</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="1">
       <widget class="QLabel" name="lab_curframe">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QSlider" name="slider_frame">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="PaintBoard" name="gView"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>947</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMENU">
    <property name="title">
     <string>MENU</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuWINDOW">
    <property name="title">
     <string>WINDOW</string>
    </property>
    <addaction name="actDockerPip"/>
   </widget>
   <widget class="QMenu" name="menuHELP">
    <property name="title">
     <string>HELP</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuTOOL">
    <property name="title">
     <string>TOOL</string>
    </property>
   </widget>
   <addaction name="menuMENU"/>
   <addaction name="menuWINDOW"/>
   <addaction name="menuHELP"/>
   <addaction name="menuTOOL"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonIconOnly</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actOpenFolder"/>
   <addaction name="actOpenFile"/>
   <addaction name="actCloseImg"/>
   <addaction name="actQuit"/>
   <addaction name="separator"/>
   <addaction name="actZoomIn"/>
   <addaction name="actZoomOut"/>
   <addaction name="actZoomRealSize"/>
   <addaction name="actFit_Width"/>
   <addaction name="actFit_Height"/>
   <addaction name="actFit_Page"/>
  </widget>
  <widget class="QDockWidget" name="dock_pipline">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>242</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>pepline</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QToolBox" name="toolBox">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="page_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>253</width>
          <height>472</height>
         </rect>
        </property>
        <attribute name="label">
         <string>run</string>
        </attribute>
        <widget class="QPushButton" name="btn_clear">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>390</y>
           <width>131</width>
           <height>29</height>
          </rect>
         </property>
         <property name="text">
          <string>clear All</string>
         </property>
        </widget>
        <widget class="QGroupBox" name="groupBox">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>0</y>
           <width>197</width>
           <height>151</height>
          </rect>
         </property>
         <property name="title">
          <string>point generator</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="2" column="0">
           <widget class="QPushButton" name="btn_execute">
            <property name="text">
             <string>execute</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>mode</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1" colspan="2">
             <widget class="QComboBox" name="combo_genseed">
              <item>
               <property name="text">
                <string>seed generate</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>draw path</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0" colspan="2">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>seed num</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLineEdit" name="edit_seed_num">
              <property name="text">
               <string>3</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="chk_draw">
            <property name="text">
             <string>draw</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>170</y>
           <width>191</width>
           <height>181</height>
          </rect>
         </property>
         <property name="title">
          <string>Delaunay triangulation</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <widget class="QPushButton" name="btn_delaunay_all">
            <property name="text">
             <string>dalaunay points</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="btn_delaunay_constrain">
            <property name="text">
             <string>constrain delaunay</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QPushButton" name="btn_delaunay_each">
            <property name="text">
             <string>dalaunay contours</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QPushButton" name="btn_clear_2">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>430</y>
           <width>131</width>
           <height>29</height>
          </rect>
         </property>
         <property name="text">
          <string>clear Result</string>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actOpenFolder">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/folder.ico</normaloff>:/images/images/folder.ico</iconset>
   </property>
   <property name="text">
    <string>Folder</string>
   </property>
   <property name="toolTip">
    <string>打开文件</string>
   </property>
  </action>
  <action name="actQuit">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/quit.png</normaloff>:/images/images/quit.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="toolTip">
    <string>退出</string>
   </property>
  </action>
  <action name="actOpenFile">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/openfile.png</normaloff>:/images/images/openfile.png</iconset>
   </property>
   <property name="text">
    <string>File</string>
   </property>
  </action>
  <action name="actZoomIn">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/zoom-in.ico</normaloff>:/images/images/zoom-in.ico</iconset>
   </property>
   <property name="text">
    <string>zoom in</string>
   </property>
   <property name="toolTip">
    <string>放大图片</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actZoomOut">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/zoom-out.ico</normaloff>:/images/images/zoom-out.ico</iconset>
   </property>
   <property name="text">
    <string>zoom out</string>
   </property>
   <property name="toolTip">
    <string>缩小图片</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actZoomRealSize">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/zoom.png</normaloff>:/images/images/zoom.png</iconset>
   </property>
   <property name="text">
    <string>resize</string>
   </property>
   <property name="toolTip">
    <string>图片实际大小显示</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
  </action>
  <action name="actDockerPip">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pipline</string>
   </property>
  </action>
  <action name="actCloseImg">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/cross.png</normaloff>:/images/images/cross.png</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actFit_Width">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/auto_fit_width_regular_icon_205619.ico</normaloff>:/images/images/auto_fit_width_regular_icon_205619.ico</iconset>
   </property>
   <property name="text">
    <string>Fit_Width</string>
   </property>
  </action>
  <action name="actFit_Height">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/auto_fit_height_regular_icon_205620.ico</normaloff>:/images/images/auto_fit_height_regular_icon_205620.ico</iconset>
   </property>
   <property name="text">
    <string>Fit Height</string>
   </property>
  </action>
  <action name="actFit_Page">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/icons8-fit-to-page-32.png</normaloff>:/images/images/icons8-fit-to-page-32.png</iconset>
   </property>
   <property name="text">
    <string>Fit Page</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PaintBoard</class>
   <extends>QGraphicsView</extends>
   <header location="global">paintboard.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
